### ISSUE TITLE
new VPN subscription count mismatching between  `moz-fx-fxa-prod-0712.fxa_prod_logs.stdout_20*`  and the table with all subscriptions for shared products

### ISSUE THREAD

**Issue by YeonjooSmith** [mozilla/fxa#11112]
## Description
The `moz-fx-fxa-prod-0712.fxa_prod_logs.stdout_20*`(I will call this, the prod table) logs event type = "fxa_pay_setup - 3ds_complete" when a user complete the VPN subscription acquisition. If the user who completes the acquisition does not cancel the subscription within a day, the subscription information of the user is saved in `moz-fx-data-shared-prod`._mozilla_vpn_derived.all_subscriptions_v1_(I will call this, the all sub table). That means that every subscription information in the all sub table should be in the prod table but that is not a case. 

### Steps to reproduce
[Here](https://console.cloud.google.com/bigquery?sq=470293616514:6b2733661a9648fc85759f4425acedec) is the query to count the daily new subscriptions from the two tables.

* in `moz-fx-data-shared-prod`.mozilla_vpn_derived.all_subscriptions_v1 for provider in( 'FxA Stripe','FxA Paypal' ) → for web purchases
* `moz-fx-fxa-prod-0712.fxa_prod_logs.stdout_20*` for event type = "fxa_pay_setup - 3ds_complete"
[This spreadsheet](https://docs.google.com/spreadsheets/d/1XBDBJFEg_SigktDyo_ynb-FctaHBnS9SJr0BZkuTP-w/edit?usp=sharing) is the result of the query linked above. 
* prod_null is the reason for this bug. prod_null is the number of new subscriptions in the all sub table that do not have the event logged in the prod table. 

### Expected result
The prod_null should be 0. 

### Actual result
over 10% of daily new subscriptions do not get logged as acquisition completed in the prod table. `prod_null_to_all_sub_uid` in [this graph](https://datastudio.google.com/reporting/fe5dd69c-2404-4ae7-b16a-d747d1503000) from [this query](
[https://console.cloud.google.com/bigquery?sq=470293616514:c4ea88af89c542a99f2ada9133cf2bd3](https://console.cloud.google.com/bigquery?sq=470293616514:c4ea88af89c542a99f2ada9133cf2bd3)
) shows the proportion of new subscriptions saved in the all sub table that do not have the event log in the prod table. 

### Environment
N/A

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-4251)

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Yeonjoo Yoo : Are these FxA tables? When I look at BigQuery, I see only moz-fx-fxa-prod-0712 for our prod BigQuery DB and I don’t see any tables under that that match moz-fx-fxa-prod-0712. Perhaps this is another team that manages this DB or Ops?

**Comment by YeonjooSmith** [comment]
@biancadanforth  I believe moz-fx-fxa-prod-0712.fxa_prod_logs.stdout_20* is the table that stores logs from fxa payment server and require a permission to access the access the data and 90 day expiration date. I cannot find the definition for the table but it is mentioned in here(https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/mozilla_vpn_derived/fxa_attribution_v1/query.sql#L10)

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Hmm that is not in the fxa repo nor in the FxA cloud ops repos I'm familiar with. In general AFAIK the FxA engineering team doesn't directly manage our BigQuery tables -- that would probably be Ops or Data ETL teams?

**Comment by data-sync-user** [comment]
➤ Jon Buckley commented:

Yeonjoo Yoo thank you for providing that query. I looked into one user account to see why it didn’t exist in the payments table: https://console.cloud.google.com/bigquery?sq=77205064629:607cf9fcd52b4e96b46ba303c4312603 

Based on the meta data in this query I tracked down the corresponding FxA payments server events: https://console.cloud.google.com/bigquery?sq=77205064629:70f5064b3ad54b128ba47da6cdd8eb93

Looking at this query, there’s no fxa uid in the logs. I guess I have two questions:
1. Why isn’t the fxa uid being emitted in these logs?

2. Where is the fxa uid that the subscriptions table has emitted from?

**Comment by data-sync-user** [comment]
➤ Yeonjoo Yoo commented:

Jon Buckley Thank you so much for looking into this issue! It seems that the fxa_uid in all subscription table  is emitted from the stripe data table according to the query(https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql ( https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql|smart-link ) ) 
Daniel Thorn  Vijay Budhram Do you have an answer or a hypothesis on the questions in Jon’s comment?

**Comment by data-sync-user** [comment]
➤ Yeonjoo Yoo commented:

Jon Buckley  I took a look at the query (https://console.cloud.google.com/bigquery?sq=77205064629:70f5064b3ad54b128ba47da6cdd8eb93). I think for this particular example, it is a new FxA user completed the payment step. When a new FxA user goes through the conversion funnel, they are supposed to  'fxa_pay_setup - 3ds_complete' appears twice with and without user id as shown in Logs in this pull request +here+ ( https://github.com/mozilla/fxa/pull/10376 ) by Vijay

  * First time without user id when the payment is completed and the email is sent to complete FxA sign up
  * Second time with user id when the sign up process is completed. 

But in this case, 'fxa_pay_setup - 3ds_complete' only appeared it once. So I looked up the flow id to see if the login process was completed(query ( https://console.cloud.google.com/bigquery?sq=470293616514:25cc7fe118994935ab321391e6c8af67 )). According to the content auth table, it showed that the log in was not completed after the registration and I think that is why  'fxa_pay_setup - 3ds_complete' did not show up with user id for the second time as it should. 

I also plotted how how many the second ‘fxa_pay_setup - 3ds_complete' are missing daily ( reason can be either users never login, FxA server missing events or something else) 

  * +plot+ ( https://sql.telemetry.mozilla.org/queries/83031/source#205882 ) refer to plot titled: new fxa user payment complete appearing twice w/o user id mismatch - possible bug
  * this might be a part of the problem but the number of cases missing the second event with user id does not match the prod null counts in the bug description.
